scripts/lib/clean_yarn_cache.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import sys
+
+ZULIP_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.append(ZULIP_PATH)
+from datetime import date, datetime
+
+from dateutil.relativedelta import relativedelta
+
+from scripts.lib.zulip_tools import may_be_perform_purging, parse_cache_script_args, run_as_root
+
+YARN_CACHE_PATH = "../.cache/yarn/"
+CURRENT_VERSION = "v6"
+
+
+def remove_unused_versions_dir(args: argparse.Namespace) -> None:
+    current_version_dir = os.path.join(YARN_CACHE_PATH, CURRENT_VERSION)
+    dirs_to_purge = set(
+        [
+            os.path.join(YARN_CACHE_PATH, directory)
+            for directory in os.listdir(YARN_CACHE_PATH)
+            if directory != CURRENT_VERSION
+        ]
+    )
+
+    may_be_perform_purging(
+        dirs_to_purge,
+        {current_version_dir},
+        "yarn cache",
+        args.dry_run,
+        args.verbose,
+        args.no_headings,
+    )
+
+
+def remove_old_files(dry_run: bool, verbose: bool) -> None:
+    if dry_run:
+        print("Performing a dry run...")
+
+    six_months_ago = date.today() + relativedelta(months=-6)
+    for path, _, files in os.walk(YARN_CACHE_PATH + CURRENT_VERSION):
+        for name in files:
+            file_path = os.path.join(path, name)
+            last_modified = os.path.getmtime(file_path)
+            # Remove file if its modification time is 6 months or more ago.
+            if datetime.date(datetime.fromtimestamp(last_modified)) <= six_months_ago:
+                if verbose:
+                    print("Cleaning unused {}: {}".format("yarn cache", file_path))
+                if not dry_run:
+                    run_as_root(["rm", "-rf", file_path])
```

Comment:
Why is this not using `may_be_perform_purging`?  I think we could reasonably call it with `dirs_to_keep` empty if that's the problem.